gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/mapper/ReferralRequestMapperTest.java

```diff
@@ -552,6 +553,63 @@ void When_MapToReferralRequest_With_NopatConfidentialityCodeWithinEhrComposition
         );
     }
 
+    @Test
+    void When_MappingReferralRequestReferencedByReferralRequestToExternalDocumentLinkSet_Expect_SupportingInfoReferencesLinksetDocuments() {
+        final RCMRMT030101UKEhrExtract ehrExtract = unmarshallEhrExtractElement(
+            "ResourceFilter",
+            "ehr_extract_with_referral_request_to_external_document_linkset.xml"
+        );
+
+        final var referralRequests = referralRequestMapper.mapResources(
+            ehrExtract,
+            (Patient) new Patient().setId(PATIENT_ID),
+            List.of(),
+            PRACTISE_CODE
+        );
+        final var referralRequest = referralRequests.getFirst();
+
+        assertAll(
+            () -> assertThat(referralRequest.getSupportingInfo())
+                .hasSize(2),
+            () -> assertThat(referralRequest.getSupportingInfo().getFirst().getReferenceElement().getResourceType())
+                .isEqualTo(ResourceType.DocumentReference.name()),
+            () -> assertThat(referralRequest.getSupportingInfo().getFirst().getReferenceElement().getIdPart())
+                .isEqualTo("narrative-statement-1"),
+            () -> assertThat(referralRequest.getSupportingInfo().getLast().getReferenceElement().getResourceType())
+                .isEqualTo(ResourceType.DocumentReference.name()),
+            () -> assertThat(referralRequest.getSupportingInfo().getLast().getReferenceElement().getIdPart())
+                .isEqualTo("narrative-statement-2")
+        );
+    }
+
+    @Test
+    void When_ReferralRequestReferencedByMultipleLinkSets_Expect_AllRelatedDocumentReferencesAddedAsSupportingInfo() {
+        final var expectedSize = 3;
+        final RCMRMT030101UKEhrExtract ehrExtract = unmarshallEhrExtractElement(
+            "ehr_extract_with_multiple_request_statement_to_external_document_linksets.xml"
+        );
+
+
+        final var referralRequests = referralRequestMapper.mapResources(
+            ehrExtract,
+            (Patient) new Patient().setId(PATIENT_ID),
+            List.of(),
+            PRACTISE_CODE
+        );
+        final var referralRequest = referralRequests.getFirst();
+        assertAll(
+            () -> assertThat(referralRequest.getSupportingInfo()).hasSize(expectedSize),
```

Comment:
Minor: The `containsExactly` below already checks that there will be three items, so this is a little redundant.